db/bigtable.go

```diff
@@ -1532,279 +1697,398 @@ func (bigtable *Bigtable) SaveValidatorIncomeDetails(epoch uint64, rewards map[u
 	return nil
 }
 
-func (bigtable *Bigtable) GetEpochIncomeHistoryDescending(startEpoch uint64, endEpoch uint64) (*itypes.ValidatorEpochIncome, error) {
-	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(time.Second*30))
-	defer cancel()
+func (bigtable *Bigtable) MigrateEpochSchemaV1ToV2(epoch uint64) error {
```

Comment:
**Question:** How exactly are you planning to call this?

This function is part of the same PR that already uses the new schema so as soon as you merge this to master, the whole explorer will already try to use the new schema even though this is also the earliest time the migration function can be executed. The explorer will not be working as long as the migration is ongoing, will it?

Also, how have you planned to handle new data that arrives while the migration is happening?